doc/guides/dts/bindings.rst

```diff
@@ -4,67 +4,79 @@ Devicetree bindings
 ###################
 
 A :ref:`devicetree` on its own is only half the story for describing the
-available hardware devices. The tree itself doesn't tell the build system
-which pieces of information are useful to device drivers, or what
-:ref:`C macros <dt-macros>` to generate from the devicetree itself.
+available hardware devices. The tree itself doesn't tell the :ref:`build system
+<build_overview>` which pieces of information are useful to :ref:`device
+drivers <device_drivers>`, or what :ref:`C macros <dt-macros>` to generate from
+the devicetree itself.
 
 *Devicetree bindings* provide the other half of this information. Zephyr
-devicetree bindings are files in YAML format. The build system only generates
-macros for devicetree nodes which have bindings.
+devicetree bindings are files in YAML format. Each binding describes the
+contents of a devicetree node in a way that lets the build system decide which
+macros to generate for it.
 
-Nodes are mapped to bindings via their ``compatible`` string(s). Take
-the following node as an example:
+.. _dt-binding-compat:
 
-.. code-block:: none
+Mapping nodes to bindings
+*************************
+
+During the :ref:`build_configuration_phase`, the build system tries to map each
+node in the devicetree to a binding file. The build system only generates
+macros for devicetree nodes which have matching bindings. Nodes are mapped to
+bindings by their :ref:`compatible properties <dt-syntax>`. Take the following
+node as an example:
+
+.. code-block:: DTS
 
    bar-device {
    	compatible = "foo-company,bar-device";
-   	...
+   	/* ... */
    };
 
-This node would get mapped to a binding with this in it:
+The build system will try to map the ``bar-device`` node to a YAML binding with
+this ``compatible:`` line:
 
 .. code-block:: yaml
 
    compatible: "foo-company,bar-device"
 
-You might also run across this legacy syntax, which works the same way:
+Built-in bindings are stored in :zephyr_file:`dts/bindings/`. Binding file
+names usually match their ``compatible:`` lines, so the above binding would be
+named :file:`foo-company,bar-device.yaml`.
 
-.. code-block:: yaml
+If a node has more than one string in its ``compatible`` property, then the
+first binding found is used, going from the first string to the last. Take this
```

Comment:
This doesn't seem quite clear.

Maybe this is a bit more clear:

`then the first ``compatible`` that matches a binding`

Reply:
Or maybe:

    then the build system looks for compatible bindings in the listed order
    and uses the first match.

Just being picky about the build system using a binding, not a "compatible" which is a string.